src/terms/terms.jl

```diff
@@ -80,7 +80,7 @@ In this case the matrix has effectively 4 blocks
 """
 @timing function compute_kernel(basis::PlaneWaveBasis{T}; kwargs...) where {T}
     n_spin = basis.model.n_spin_components
-    kernel = zeros(T, n_spin * prod(basis.fft_size), n_spin * prod(basis.fft_size))
+    kernel = zeros_like(basis.G_vectors, T, n_spin * prod(basis.fft_size), n_spin * prod(basis.fft_size))
```

Comment:
The line now becomes a bit long. Maybe cache the `prod(basis.fft_size)` or the `n_spin * prod(basis.fft_size)` in a variable and use it in the line thereafter for clarity.

Reply:
Good point, updated now.